libs/design-system-core/src/utils/get-cropped-image.ts

```diff
@@ -26,11 +26,22 @@ const rotateSize = (width: number, height: number, rotation: number) => {
  */
 const createImage = (url: string): Promise<HTMLImageElement> =>
   new Promise((resolve, reject) => {
```

Comment:
no need to wrap into a new promise, fetch is already returning a promise. you can also use async/await.
how is this resolving the cors issue?

Reply:
- the promise is there to wait for the image to load, then to resolve with the image otherwise if during loading there is an error to reject with the error ... and also during fetching the url if there is an issue the catch will reject the promise
- the previous approach required to set crossOrigin attribute on the image element and the options where limited which resulted in the opaque response because there was a mismatch in how the server expected to handle the response and how the request was made ... but now when using fetch the default mode is to use cors and this time there was no mismatch

Reply:
cors should be mandatory for loading content from 3rd party servers

Reply:
the proposed solution could be replaced by removing just 1LOC `image.setAttribute('crossOrigin', 'anonymous');`

Reply:
but if this approach has better results we should observe for possible issues on next

Reply:
you could even return just the `ObjectURL` since this one gets transformed back again from the image object

Reply:
and `revokeObjectURL` inside the callback function from canvas.toBlob

Reply:
actually removing that one line will throw an error when trying to access the canvas image data ... it gives an error saying canvas has been tainted by cross-origin data but the fetch approach is clean imo we can observe the behavior on next ... yes I could return the object but I need the image object to access the dimensions and such info